docling_core/types/doc/document.py

```diff
@@ -1683,6 +1775,371 @@ def transform_to_content_layer(cls, data: dict) -> dict:
                     item["content_layer"] = "furniture"
         return data
 
+    # ---------------------------
+    # Public Manipulation methods
+    # ---------------------------
+
+    def append_child_item(
+        self, *, child: NodeItem, parent: Optional[NodeItem] = None
+    ) -> RefItem:
```

Comment:
Let's just return `None` and not return the reference str, as the latter is static.

The user anyway holds (a pointer to) the `child` object, so at any point they can access its current reference with `child.get_ref()`.

Same applies for all public methods.